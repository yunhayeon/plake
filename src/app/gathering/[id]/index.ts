import GatheringDetailPage from "./page";

export default GatheringDetailPage;
