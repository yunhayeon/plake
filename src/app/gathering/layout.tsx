import CreateGatheringModal from "@/components/modals/create-gathering-modal/CreateGatheringModal";

const GatheringLayout = ({ children }: { children: React.ReactNode }) => {
  return (
    <>
      {children}
      <CreateGatheringModal />
    </>
  );
};

export default GatheringLayout;
