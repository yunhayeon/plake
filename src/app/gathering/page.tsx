import { Suspense } from "react";

import GatheringFilterSort from "@/components/common/GatheringFilterSort";
import MainCardItem from "@/components/layout/MainCardItem";
import SubTab from "@/components/navigations/SubTab";
import Tab from "@/components/navigations/Tab";

const cardData = {
  id: 0,
  name: "달램핏 오피스 스트레칭",
  dateTime: new Date(),
  registrationEnd: new Date(),
  location: "강남구",
  participantCount: 6,
  capacity: 20,
  image: "https://picsum.photos/200/300",
};

const subTabData = ["전체", "운동", "예술", "미식", "기타"];

const Page = () => {
  return (
    <div>
      <div className="base-wrap">
        <Suspense>
          <Tab />
        </Suspense>
        <SubTab tabList={subTabData} />
        <GatheringFilterSort />
        <MainCardItem
          key={cardData.id}
          name={cardData.name}
          dateTime={cardData.dateTime}
          registrationEnd={cardData.registrationEnd}
          location={cardData.location}
          participantCount={cardData.participantCount}
          capacity={cardData.capacity}
          image={cardData.image}
        />
      </div>
    </div>
  );
};

export default Page;
