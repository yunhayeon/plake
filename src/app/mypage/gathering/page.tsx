import EmptyState from "@/app/mypage/_components/EmptyState";

const Page = () => {
  return (
    <>
      {/* 추가 예정 */}
      <EmptyState message="아직 만든 모임이 없어요" />
    </>
  );
};

export default Page;
