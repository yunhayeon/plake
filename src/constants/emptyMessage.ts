export const EMPTY_MESSAGE = {
  mypage: {
    default: "신청한 모임이 아직 없어요",
    reviews: "아직 작성 가능한 리뷰가 없어요",
    written: "아직 작성한 리뷰가 없어요",
    gatherings: "아직 만든 모임이 없어요",
  },
};
