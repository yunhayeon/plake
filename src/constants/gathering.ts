export const GATHERING = {
  CAPACITY: {
    MIN: 5,
    MAX: 20,
  },
} as const;
