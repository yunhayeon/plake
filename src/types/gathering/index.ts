import { IUser } from "@/types/user";

export enum GatheringType {
  MINDFULNESS = "MINDFULNESS",
  DALLAEMFIT = "DALLAEMFIT",
  OFFICE_STRETCHING = "OFFICE_STRETCHING",
  WORKATION = "WORKATION",
}

export interface IGathering {
  teamId: string;
  id: number;
  name: string;
  type: GatheringType;
  dateTime: string;
  registrationEnd: string;
  location: string;
  participantCount: number;
  capacity: number;
  image: string | null;
  createdBy: number;
  canceledAt: string | null;
}

export interface IParticipant {
  User: IUser;
  userId: number;
  teamId: string;
  gatheringId: number;
  joinedAt: string;
}
