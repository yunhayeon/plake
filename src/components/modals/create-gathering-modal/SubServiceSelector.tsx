import clsx from "clsx";

import { Button } from "@/components/ui/Button";
import { SUB_SERVICE_LIST } from "@/constants/gathering";

interface ISubServiceSelectorProps {
  isOffline: boolean;
  selectedSubService?: string;
  onClickTab: (value: string) => void;
}

const SubServiceSelector = ({
  isOffline,
  selectedSubService,
  onClickTab,
}: ISubServiceSelectorProps) => {
  return (
    <div className="flex w-full gap-2">
      {SUB_SERVICE_LIST[isOffline ? "OFFLINE" : "ONLINE"].map(subService => (
        <Button
          className={clsx(
            "w-full bg-gray-200 text-sm font-medium text-gray-900 hover:bg-opacity-90",
            selectedSubService === subService.value && "bg-gray-900 text-white",
          )}
          variant="default"
          key={subService.value}
          onClick={() => onClickTab(subService.value)}
        >
          {subService.name}
        </Button>
      ))}
    </div>
  );
};

export default SubServiceSelector;
