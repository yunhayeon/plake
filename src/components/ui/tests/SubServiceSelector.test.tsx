import { fireEvent, render, screen } from "@testing-library/react";

import SubServiceSelector from "@/components/ui/SubServiceSelector";
import { SUB_SERVICE_LIST } from "@/constants/gathering";

describe("SubServiceSelector", () => {
  const mockOnClickTab = jest.fn();
  const subServices = SUB_SERVICE_LIST.OFFLINE;

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("컴포넌트가 정상적으로 렌더링 되어야 함", () => {
    render(
      <SubServiceSelector selectedSubService="" onClickTab={mockOnClickTab} />,
    );

    subServices.forEach(subService => {
      expect(
        screen.getByRole("button", { name: subService.name }),
      ).toBeInTheDocument();
    });
  });

  it("버튼을 클릭하면 onClickTab이 올바르게 호출됨", () => {
    render(
      <SubServiceSelector selectedSubService="" onClickTab={mockOnClickTab} />,
    );

    const firstButton = screen.getByRole("button", {
      name: subServices[0].name,
    });
    fireEvent.click(firstButton);

    expect(mockOnClickTab).toHaveBeenCalledTimes(1);
    expect(mockOnClickTab).toHaveBeenCalledWith(subServices[0].value);
  });

  it("선택된 버튼에 올바른 스타일이 적용됨", () => {
    render(
      <SubServiceSelector
        selectedSubService={subServices[0].value}
        onClickTab={mockOnClickTab}
      />,
    );

    const selectedButton = screen.getByRole("button", {
      name: subServices[0].name,
    });
    expect(selectedButton).toMatchSnapshot();
  });
});
