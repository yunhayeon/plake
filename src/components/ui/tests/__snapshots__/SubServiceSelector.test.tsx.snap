// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubServiceSelector 선택된 버튼에 올바른 스타일이 적용됨 1`] = `
<button
  class="flex items-center justify-center whitespace-nowrap rounded-md transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 shadow h-9 px-4 py-2 w-full text-sm font-medium hover:bg-opacity-90 bg-gray-900 text-white"
  role="button"
  type="button"
>
  운동
</button>
`;
